### PR TITLE
tests: verify multiple nameservers config via kernel arguments work well

### PR DESCRIPTION
**Pull request by HuijingHei**

### CODE REVIEW

File: tests/kola/networking/nameserver [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+set -xeuo pipefail
+
+. $KOLA_EXT_DATA/commonlib.sh
+
+# check nameserver in /etc/resolv.conf
+resolv=/etc/resolv.conf
+if ! (grep -q "nameserver 8.8.8.8" ${resolv} && \
+      grep -q "nameserver 1.1.1.1" ${resolv}); then
+    fatal "Error: can not find nameserver in ${resolv}"
```
Comment (jlebon) [inline tests/kola/networking/nameserver]: Can we `cat ${resolv}` before this line so we can tell what the actual file looked like?

File: tests/kola/networking/nameserver [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+# check nameserver in config file
+conf=/etc/NetworkManager/system-connections/default_connection.nmconnection
+if ! grep -q "dns=8.8.8.8;1.1.1.1;"; then
+    fatal "Error: can not find nameserver in ${conf}"
```
Comment (jlebon) [inline tests/kola/networking/nameserver:36]: Similarly here.

File: tests/kola/networking/nameserver [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+ok "multipule nameserver"
```
Comment (jlebon) [inline tests/kola/networking/nameserver]: Minor: multiple

**Review by jlebon — DISMISSED** [review]
That's a nice test idea. Minor nits, but LGTM overall. I opened https://github.com/coreos/fedora-coreos-config/pull/2032 for the CI failure.

**Comment by dustymabe** [comment]
This test isn't being tested in CI:

```
kola: Found non-executable file with shebang: nameserver
```

You need to make it executable in git with a `chmod +x file && git add file`.

File: tests/kola/networking/nameserver [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#   - This test should pass everywhere if it passes anywhere.
+
+set -xeuo pipefail
+
+. $KOLA_EXT_DATA/commonlib.sh
+
+# check nameserver in /etc/resolv.conf
+resolv=/etc/resolv.conf
+if ! (grep -q "nameserver 8.8.8.8" ${resolv} && \
+      grep -q "nameserver 1.1.1.1" ${resolv}); then
```
Comment (dustymabe) [inline tests/kola/networking/nameserver]: I don't think this test is going to work on FCOS where we have systemd-resolved. Maybe we can modify the test to work on both FCOS and RHCOS. on FCOS you should be able to run `resolvectl` to see what nameservers are configured.

**Review by dustymabe — CHANGES_REQUESTED** [review]
Found a few things that need to be changed.

File: tests/kola/networking/nameserver (hunk 4 above)
Comment (dustymabe) [reply tests/kola/networking/nameserver]: This isn't resolved. See the test failure in CI.

**Comment by HuijingHei** [comment]
> Found a few things that need to be changed.

Done, thanks!

File: tests/kola/networking/nameserver [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        grep -q "nameserver 1.1.1.1" ${resolv}); then
+        fatal "Error: can not find nameserver in ${resolv}"
+    fi
+
+    # check nameserver in config file
+    conf=/etc/NetworkManager/system-connections/default_connection.nmconnection
+    cat ${conf}
+    if ! grep -q "dns=8.8.8.8;1.1.1.1;" ${conf}; then
+        fatal "Error: can not find nameserver in ${conf}"
+    fi
```
Comment (dustymabe) [inline tests/kola/networking/nameserver]: I think this part can be checked on both RHCOS and FCOS, right?

**Review by cgwalters — APPROVED** [review]

**Review by dustymabe — APPROVED** [review]

**Pull request merged**